omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,236 @@
+		String json = "{\"allergen\": {\"allergenType\":\"DRUG\", \"codedAllergen\": { \"uuid\": \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\"}, \"nonCodedAllergen\": \"test non coded\" }}";
+		SimpleObject newAllergy = deserialize(handle(newPostRequest(getURI(), json)));
+		
+		Assert.assertEquals("DRUG", Util.getByPath(newAllergy, "allergen/allergenType"));
+		Assert.assertEquals("test non coded", Util.getByPath(newAllergy, "allergen/nonCodedAllergen"));
+		Assert.assertEquals("5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA", Util.getByPath(newAllergy, "allergen/codedAllergen/uuid"));
+		allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 5);
+	}
+
+	/**
+	 *  Set No Known Allergies icorrectly using empty post body
+	 *  throw ResourceDoesNotSupportOperationException because Patient does not exist
+	 */
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldThrowExceptionWhenSettingNoKnownAllergiesForNonExistingPatientUsingEmptyPostBody() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);		
+		Assert.assertEquals(Allergies.SEE_LIST, allergies.getAllergyStatus());
+
+		String nonExistentPatientURI = "patient/nonExistingPatient/allergy";
+		String json = "{}";
+		handle(newPutRequest(nonExistentPatientURI, json));
+	}
+	
+	/**
+	 *  Set No Known Allergies icorrectly using non empty post body and List of Allergies is empty
+	 *  throw ResourceDoesNotSupportOperationException because of non empty post body
+	 */
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldThrowExceptionWhenSettingNoKnownAllergiesWithExistingPatientUsingNonEmptyPostBody() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);		
+		Assert.assertEquals(Allergies.SEE_LIST, allergies.getAllergyStatus());
+		
+		// delete existing allergies
+		handle(newDeleteRequest(getURI(), 
+				new Parameter("reason", "unit test")));
+
+		allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+
+		String json = "{\"allergen\": {\"allergenType\":\"DRUG\", \"nonCodedAllergen\": \"test non coded\" }}";
+		handle(newPutRequest(getURI(), json));
+	}
+	
+	/**
+	 *  Set No Known Allergies icorrectly using empty post body
+	 *  throw ResourceDoesNotSupportOperationException because List of Allergies is not empty
+	 */
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldThrowExceptionWhenSettingNoKnownAllergiesWithExistingPatientUsingEmptyPostBody() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());

```

Comment:
If they are multiple, which of them is this particular test dealing with? It should be reflected in the method name.
Otherwise just saying that it should throw an exception when setting no known allergies with existing patient is confusing because there are times when an exception should not be thrown when setting no known allergies with an existing patient.